panel/widgets/slider.py

```diff
@@ -191,14 +264,38 @@ def _process_property_change(self, msg):
 
 
 class DiscreteSlider(CompositeWidget, _SliderBase):
+    """The DiscreteSlider widget allows selecting a value from a discrete list or dictionary of
+    values using a slider.
+
+    Reference: https://panel.holoviz.org/reference/widgets/DiscreteSlider.html
 
-    options = param.ClassSelector(default=[], class_=(dict, list))
+    ## Example
 
-    value = param.Parameter()
+    ```python
+    import panel as pn
+
+    pn.extension()
+
+    slider = pn.widgets.DiscreteSlider(
+        value="love",
+        options=list(["We", "love", "data", "apps"]),
```

Comment:
Maybe use numbers instead of strings in this example, it feels like a more natural usage of a discrete slider to me.

Reply:
Should either use `list` or `[]`